src/high-level/abstract-tensor.lisp

```diff
@@ -309,3 +309,9 @@ If TARGET is not specified then a new tensor is created with the same element ty
               tensor
               target)
       target)))
+
+(declaim (inline make-array))
+(defun make-array (dimensions &rest args &key &allow-other-keys)
+  (apply #'cl:make-array dimensions
+         #+lispworks8 :allocation :pinnable
```

Comment:
I think you need two reader conditionals here: `#+lw :allocation #+lw ':pinnable` (abbreviated)